### PR TITLE
Adding the ability to use cache-manager from Local predictors

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

**Comment by mmerce** [comment]
Please, don't merge till the docs explaining its use are added.

**Comment by mmerce** [comment]
I already added the info in docs and tests passed, so it's ready for review

File: docs/index.md [hunk 1]
```diff
@@ -967,6 +967,83 @@ we asked for the 2 most recent sources:
...
+
+Every model available in BigML is white-box and can be downloaded from the
+API as a JSON. This response can either be stored in the file system as a file
+or stored in a cache-manager. In both cases, the bindings provide a class
+for every model which can interpret this JSON and predict (supervised models),
+or assing centroids, compute anomaly scores, etc. Thus, these classes allow
+you to use your BigML model locally, with no connection whatsoever to
+BigML's servers.
+
+The following sections explain each of these classes and its methods. As a
```
Comment (jaor) [inline docs/index.md]: its -> their

File: lib/BigML.js [hunk 2]
```diff
@@ -160,24 +169,45 @@ BigML.prototype.request = function (options, retry, cb) {
...
+            // defined in the cache manager constructor
+            self.storage.set(safeId, body,
+                             function(error) {
+              if (error) {logger.error('Problem caching ' + options.resource +
+                                       ': ' + error);}
+              if (process.env.BIGML_DEBUG || false) {
+                self.storage.get(safeId, function (error, data) {
+                  logger.info("Stored in cache:"+ JSON.parse(data).resource)});}});
+          } else {
+            logger.error("Storage can the path to a file system directory" +
```
Comment (jaor) [inline lib/BigML.js]: typo: can be

File: lib/BigML.js [hunk 3]
```diff
@@ -160,24 +169,45 @@ BigML.prototype.request = function (options, retry, cb) {
...
+                         " used as storage or to a cache-manager object" +
+                         " that provides set, get and del methods. " +
+                         "None of these options has been found.");
```
Comment (jaor) [inline lib/BigML.js]: nit.  I would say: "Invalid storage.  Storage can be .... del methods." (and omit the last sentence)

File: lib/LocalEnsemble.js [hunk 4]
```diff
@@ -322,26 +312,62 @@ function LocalEnsemble(ensembleOrModels, connection, maxModels) {
...
+    self.on('ensembleReady', fillSeparateModelsInfo);
+
+    filename = null;
+    if (typeof resource === 'string') {
+      // When the argument contains the path to a JSON file
+      if (fs.existsSync(resource)) {
+        filename = resource;
+      } else if (typeof this.connection.storage === 'string') {
+      // Loads from file in local fs directory
+      filename = utils.getStoredModelFile(resource, this.connection);
```
Comment (jaor) [inline lib/LocalEnsemble.js]: i think the indentation is off here

File: lib/utils.js [hunk 5]
```diff
@@ -854,11 +859,18 @@ exports = {
   },
 
   getStoredModelFile: function (resourceId, connection) {
+  /**
+   * Tries to read the JSON structure from a stored file
+   *
```
Comment (jaor) [inline lib/utils.js]: and what happens if it can't? we could mention it here...

File: lib/utils.js [hunk 6]
```diff
@@ -868,13 +880,107 @@ exports = {
                              resource.replace("/", "_"));
         if (fs.existsSync(filename)) {
           return filename;
-        } else {
-          return null;
         }
+        return null;
       } catch (err) { console.log("Stored file error:", err);}
```
Comment (jaor) [inline lib/utils.js]: hmm, i know this was already here, but it's not normal to log in libraries... if we do it elsewhere, it's okay here too.

**Pull request merged**